custom/src/main/res/navigation/custom_nav_graph.xml

```diff
@@ -24,6 +24,12 @@
     android:id="@+id/customReaderFragment"
     android:name="org.kiwix.kiwixmobile.custom.main.CustomReaderFragment"
     android:label="CustomReaderFragment">
+    <action
+      android:id="@+id/action_customReader_to_customDownload"
+      app:destination="@id/customDownloadFragment"
+      app:popUpTo="@id/customDownloadFragment"
```

Comment:
not the correct popUpTo, You want to see `popUpTo` to "get rid of everything above LibraryFragment" and then with `inclusive` say "also get rid of LibraryFragment"

Reply:
Nicee, now I can see things coming together, thanks!

Reply:
Okay, so whether I popUpTo downloadFrag with inclusive false or to `customReader` with inclusive `true`, it doesn't work, instead crashes. How can this happen when I'm clearing stack till reader with reader inclusive and there is nothing left other than download?

Reply:
it is possible you are not regenerating the navigation action and using the previous one, perhaps do a clean and uninstall before installing. popUpTo Reader inclusive true is what you want reader->download, popUpTo Download inclusive true is what you want for download->reader. If that isn't working post the trace

Reply:
Actually, there are a couple of problems occurring, with the very first launch of _custom_, after granting permission, it is unable to send it successfully and therefore contributes to error and for some reason it is unable to find the action that navigates to download(both of these take place on very first launch only).
|Crash on Launch logs-I|
|--|
|![crashOnLaunch](https://user-images.githubusercontent.com/56556919/100866687-0c9dd400-34bf-11eb-9b75-6b43bf218665.png)|

|Crash on Launch logs-II|
|--|
|![crashOnLaunch-II](https://user-images.githubusercontent.com/56556919/100866713-14f60f00-34bf-11eb-943c-e63e3e21e9a9.png)|


* And second one occurs on the successive launches, when app is restarted and back press is triggered from the download screen, in this it says customReader isn't attached to an activity:

|Crash on backPress at download screen|
|--|
|![crashOnBackPress](https://user-images.githubusercontent.com/56556919/100866925-64d4d600-34bf-11eb-941f-f1492bbfb071.png)|

Reply:
Look to the PR I did, if the principle works there it should work here, the design of coreReader is much much worse & complex I will grant you, it is very possible code is executed outside of normal lifecycle flows so you may have to safeguard against it being executed by wrapping code in `if (isAdded())`

In instance 1 the error is you are at custom download trying to navigate to custom download, this is illegal as the action to do that is on Reader. Same for error 2.
Error 3 is code executing in CoreReader while it is not attached to any activity

Reply:
* That pops weren't working because I was using wrong Id in that for that param, it was the mistake of a `+`, no clue how it slipped off my mind.
* Now as I analyse the crash at first launch, I see that crash occurs just after I grant storage permission, so as soon as I click on it, the permission is granted, the reader is relaunch and at the sametime it passes the result info to the activity and before this happens, the customReader comes to screen and immediately goes to download, resulting in failure of proper transmission of premissions result. I was trying to fix it by adding a delay of few ms but it didn't let me use coroutines there

Reply:
These crashes have been fixed via latest commits pushed after [this review request](https://github.com/kiwix/kiwix-android/pull/2371#event-4095361508).